posts/2024-03-18_xportr_0_4_0/xportr_0_4_0.qmd

```diff
@@ -0,0 +1,112 @@
+  Val = c("1", "2", "3"),
+  Param = c("param1", "param2", "param3")
+)
+```
+
+and the define metadata information
+```{r,  message = FALSE, warnings = FALSE}
+metadata <- data.frame(
+  dataset = "test",
+  variable = c("Subj", "Param", "Val", "NotUsed"),
+  type = c("numeric", "character", "numeric", "character"),
+  format = NA,
+  order = c(1, 3, 4, 2)
+)
+```
+
+Users can use the [`xportr_metadata()`](https://atorus-research.github.io/xportr/reference/xportr_metadata.html) function along with [`xportr_type()`](https://atorus-research.github.io/xportr/reference/xportr_type.html) and [`xportr_order()`](https://atorus-research.github.io/xportr/reference/xportr_order.html) to not only apply metadata attributes to your data, receive feedbacks on your data based on your spec, all while also performing valuable checks.
+
+```{r, warnings = FALSE}
+adsl %>%
+  xportr_metadata(metadata, "test") %>%
+  xportr_type() %>%
+  xportr_order()
+```
+
+You can also use additional functions like [`xportr_write()`](https://atorus-research.github.io/xportr/reference/xportr_write.html) to generate your `xpt` file.
+
+# Key Features of [`xportr`](https://atorus-research.github.io/xportr/):
+
+- **Efficient Handling**: [`xportr`](https://atorus-research.github.io/xportr/) offers efficient avenues to produce `xpt` files, allowing users to manipulate clinical datasets with ease and export in `xpt` format in R.
+- **Data Integrity**: [`xportr`](https://atorus-research.github.io/xportr/) preserves data integrity by retaining variable labels, formats, and other metadata during data export operations.
+- **User-Friendly Interface**: [`xportr`](https://atorus-research.github.io/xportr/) provides a user-friendly interface with intuitive functions for generating `xpt` datasets and performing data manipulation tasks.
+
+# About the other kid on the blog, (`JSON`)! Is The Future `XPT` or JSON?
+
+`XPT` datasets are standardized, structured files used for storing clinical trial data, making them essential for regulatory submissions and data analysis. They remain indispensable in certain industries and with advancements in data management and analysis tools, the future of `xpt` datasets lies in improved efficiency, compatibility, and ensuring seamless integration with emerging technologies and data formats.
+
+As data continues to play a central role in decision-making and innovation, the future of JavaScript Object Notation (JSON) datasets are worth exploring. JSON datasets offer flexibility, simplicity, and interoperability, making them increasingly popular for data storage and exchange. JSON's lightweight and human-readable format make it suitable for a wide range of applications, including web development, APIs, IoT, and data storage. While `xpt` datasets remain prevalent in certain industries, there's a growing interest in JSON datasets across various domains. Although we will not be exploring JSON datasets, this formatted data type was worth mentioning in this blog.
```

Comment:
```suggestion
As data continues to play a central role in decision-making and innovation, the future of JavaScript Object Notation (JSON) datasets is worth exploring. JSON datasets offer flexibility, simplicity, and interoperability, making them increasingly popular for data storage and exchange. JSON's lightweight and human-readable format make it suitable for a wide range of applications, including web development, APIs, IoT, and data storage. While `xpt` datasets remain prevalent in certain industries, there's a growing interest in JSON datasets across various domains. Although we will not be exploring JSON datasets, this formatted data type was worth mentioning in this blog.
```

I have heard about JSON being discussed as a standard data format in the industry before. Would you be able to link here some further reading on this?